tsdb/engine/tsm1/engine.go

```diff
@@ -390,12 +390,7 @@ func (e *Engine) addToIndexFromKey(shardID uint64, key string, fieldType influxq
 	m := index.CreateMeasurementIndexIfNotExists(measurement)
 	m.SetFieldName(field)
 
-	mf := e.measurementFields[measurement]
-	if mf == nil {
-		mf = tsdb.NewMeasurementFields()
-		e.measurementFields[measurement] = mf
-	}
-
+	mf := e.MeasurementFields(measurement)
```

Comment:
eugh, it could if the measurement field doesn't exist. 

Since `addToIndexFromKey` is sometimes called while under an `RLock` and sometimes isn't,  this is going to take a little re-working. I'll push something up tomorrow